packages/tangle/attachment.go

```diff
@@ -0,0 +1,161 @@
+package tangle
+
+import (
+	"fmt"
+
+	"github.com/iotaledger/goshimmer/packages/ledgerstate"
+	"github.com/iotaledger/hive.go/byteutils"
+	"github.com/iotaledger/hive.go/marshalutil"
+	"github.com/iotaledger/hive.go/objectstorage"
+	"github.com/iotaledger/hive.go/stringify"
+)
+
+// Attachment stores the information which transaction was attached by which message. We need this to be able to perform
+// reverse lookups from transactions to their corresponding messages that attach them.
+type Attachment struct {
+	objectstorage.StorableObjectFlags
+
+	transactionID ledgerstate.TransactionID
+	messageID     MessageID
+}
+
+// NewAttachment creates an attachment object with the given information.
+func NewAttachment(transactionID ledgerstate.TransactionID, messageID MessageID) *Attachment {
+	return &Attachment{
+		transactionID: transactionID,
+		messageID:     messageID,
+	}
+}
+
+// AttachmentFromBytes unmarshals an Attachment from a sequence of bytes - it either creates a new object or fills the
+// optionally provided one with the parsed information.
+func AttachmentFromBytes(bytes []byte) (result *Attachment, consumedBytes int, err error) {
+	marshalUtil := marshalutil.New(bytes)
+	result, err = ParseAttachment(marshalUtil)
+	consumedBytes = marshalUtil.ReadOffset()
+
+	return
+}
+
+// ParseAttachment is a wrapper for simplified unmarshaling of Attachments from a byte stream using the marshalUtil package.
+func ParseAttachment(marshalUtil *marshalutil.MarshalUtil) (result *Attachment, err error) {
+	result = &Attachment{}
+	if result.transactionID, err = ledgerstate.TransactionIDFromMarshalUtil(marshalUtil); err != nil {
+		err = fmt.Errorf("failed to parse transaction ID in attachment: %w", err)
+		return
+	}
+	if result.messageID, err = MessageIDFromMarshalUtil(marshalUtil); err != nil {
+		err = fmt.Errorf("failed to parse message ID in attachment: %w", err)
+		return
+	}
+
+	return
+}
+
+// AttachmentFromObjectStorage gets called when we restore an Attachment from the storage - it parses the key bytes and
+// returns the new object.
+func AttachmentFromObjectStorage(key []byte, data []byte) (result objectstorage.StorableObject, err error) {
+	result, _, err = AttachmentFromBytes(byteutils.ConcatBytes(key, data))
+	if err != nil {
+		err = fmt.Errorf("failed to parse attachment from object storage: %w", err)
+	}
+
+	return
+}
+
+// TransactionID returns the transactionID of this Attachment.
+func (attachment *Attachment) TransactionID() ledgerstate.TransactionID {
```

Comment:
```suggestion
func (a *Attachment) TransactionID() ledgerstate.TransactionID {
```

Only use the first letter as a receiver name everywhere.